jcl/src/java.base/share/classes/jdk/internal/vm/Continuation.java

```diff
@@ -166,10 +168,8 @@ public final void run() {
 			} else {
 				/* Continuation yielded */
 			}
+			JLA.setContinuation(carrier, parent);
```

Comment:
Do we need to re-derive the carrier thread in case the continuation is yielded and resumed?
```suggestion
			JLA.setContinuation(JLA.currentCarrierThread(), parent);
```

Reply:
no, since yield will have to return to here as if it completed so the carrier thread cannot change. when resuming on different carrier thread, the Continuation have to call `run()` again to mount which will refetch the carrier